incubator/hnc/docs/user-guide/how-to.md

```diff
@@ -176,78 +176,80 @@ Kubernetes 1.15 and higher installed. See [issue
 In Kubernetes 1.14 and earlier, HNC is unable to stop you from deleting
 namespaces.**
 
-You cannot delete a subnamespace by deleting its namespace:
+Subnamespaces are _always_ manipulated via their anchors. For example, you
+cannot delete a subnamespace by deleting it directly:
 
 ```
 $ kubectl delete namespace child
 # Output:
-Error from server (Forbidden): admission webhook "vnamespace.k8s.io" denied the request: The namespace "child" is a subnamespace. Please delete the subnamespace anchor from the parent namespace "parent" instead.
+Error from server (Forbidden): admission webhook "namespaces.x-hnc.k8s.io" denied the request: The namespace "child" is a subnamespace. Please delete the subnamespace anchor from the parent namespace "parent" instead.
 ```
 
-However, if you simply try to delete the subnamespace anchor, it will give you a validation error:
+Instead, you must delete its anchor (note that `subns` is a short form of
+`subnamespaceanchor`):
 
 ```
-$ kubectl delete subns -nparent child
-# Output:
-Error from server (Forbidden): admission webhook "subnamespaces.hnc.x-k8s.io" denied the request: The subnamespace child doesn't allow cascading deletion. Please set allowCascadingDelete flag first.
+$ kubectl delete subns child -n parent
 ```
 
-_Note: `subns` is a short form of `subnamespaceanchor`._
-
-Similarly, if you try to delete the parent of a subnamespace, you’ll get a
-validation error, even if the parent is not a subnamespace itself:
+This _seems_ to imply that if you delete a _parent_ namespace, all its
+subnamespace children (and their descendants) will be deleted as well, since all
+objects in a namespace (such as anchors) are deleted along with the namespace.
+However, if you actually try this, you'll get an error:
 
 ```
 $ kubectl delete namespace parent
 # Output:
-Error from server (Forbidden): admission webhook "vnamespace.k8s.io" denied the request: Please set allowCascadingDelete first either in the parent namespace or in all the subnamespaces.
+Error from server (Forbidden): admission webhook "namespaces.hnc.x-k8s.io" denied the request: Please set allowCascadingDelete first either in the parent namespace or in all the subnamespaces.
  Subnamespace(s) without allowCascadingDelete set: [child].
 ```
 
 These errors are there for your protection. Deleting namespaces is very
 dangerous, and deleting _subnamespaces_ can result in entire subtrees of
-namespaces being deleted as well. Therefore, you may set the
-`allowCascadingDelete` field either on the child namespace, on its parent, or (if
-the parent is a subnamespace itself) on its parent, and so on. For example, set
-the field on `child` if you’re only trying to delete the child, or on `parent` if
-you’re trying to delete the parent.
+namespaces being deleted as well. Therefore, if deleting a namespace (or
+subnamespace) would result in the deletion of any namespace _other than the one
+explicitly being deleted_, HNC requires that you must specify the
```

Comment:
This is documented well but I am just thinking about the easy deletion of leaf subns. Is it possible that users may delete a leaf subns' parent (full ns) without knowing the subns will be deleted too?

Reply:
That's not allowed:

```
aludwin@aludwin0:hnc$ k create ns parent
namespace/parent created

aludwin@aludwin0:hnc$ k hns create child -n parent
Successfully created "child" subnamespace anchor in "parent" namespace

aludwin@aludwin0:hnc$ k delete ns parent
Error from server (Forbidden): admission webhook "namespaces.hnc.x-k8s.io" denied the request: Please set allowCascadingDelete first either in the parent namespace or in all the subnamespaces.
  Subnamespace(s) without allowCascadingDelete set: [child].
```

Reply:
Ah right, thank you for clarification. Yes, I tested it and got the same. Only deleting the leaf subns anchor doesn't need allowCascadingDelete set.